pkg/controller/pkg/manager/revisioner.go

```diff
@@ -63,15 +63,11 @@ func (r *PackageRevisioner) Revision(ctx context.Context, p v1alpha1.Package) (s
 	if err != nil {
 		return "", err
 	}
-	img, err := r.fetcher.Fetch(ctx, ref, v1alpha1.RefNames(p.GetPackagePullSecrets()))
-	if err != nil {
+	d, err := r.fetcher.Head(ctx, ref, v1alpha1.RefNames(p.GetPackagePullSecrets()))
+	if err != nil || d == nil {
```

Comment:
In what circumstance would `d` and `err` both be `nil`? If this is only covering the `NopFetcher`, perhaps that could return `&v1.Descriptor{}, nil` instead?

Reply:
@negz they should not ever both be `nil`. However, `Head` does return a `*v1.Descriptor` so I think we should check for `nil` regardless even if in practice it should never occur